Main build failing with 'RuntimeWarning: overflow encountered in scalar subtract'

Looks like it may be due to stricter checks in NumPy 2.0.0.

```
tau_p = np.uint8(0), lambda_p = np.float64(0.0), ploidy_p = np.uint64(2)
kinship_pp = 0.0, tau_q = np.uint8(2), lambda_q = np.float64(0.167)
ploidy_q = np.uint64(4), kinship_qq = np.float64(0.307798800184587)
kinship_pq = 0.0

    @numba_jit
    def _hamilton_kerr_inbreeding_non_founder(
        tau_p: int,
        lambda_p: float,
        ploidy_p: int,
        kinship_pp: float,
        tau_q: int,
        lambda_q: float,
        ploidy_q: int,
        kinship_qq: float,
        kinship_pq: float,
    ) -> float:  # pragma: no cover
        """Calculate the inbreeding coefficient of a non founder
        individual i with parents p and q.
        """
        pat = (
            tau_p
>           * (tau_p - 1)
            * (lambda_p + (1 - lambda_p) * ((ploidy_p * kinship_pp - 1) / (ploidy_p - 1)))
        )
E       RuntimeWarning: overflow encountered in scalar subtract
```

From https://github.com/sgkit-dev/sgkit/actions/runs/9540232291/job/26291678469

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -12,7 +12,7 @@ scikit-learn
 partd
 bed-reader
 rechunker
-cbgen; platform_system != "Windows"
+cbgen < 1.0.5; platform_system != "Windows"
 cbgen == 1.0.1; platform_system == "Windows"
 cyvcf2; platform_system != "Windows"
 yarl

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,4 @@
-numpy
+numpy < 2
 xarray
 dask[array] >= 2023.01.0
 distributed >= 2023.01.0

--- a/setup.cfg
+++ b/setup.cfg
@@ -27,7 +27,7 @@ zip_safe = False  # https://mypy.readthedocs.io/en/latest/installed_packages.htm
 include_package_data = True
 python_requires = >=3.9
 install_requires =
-    numpy
+    numpy < 2
     xarray
     dask[array] >= 2022.01.0
     distributed >= 2022.01.0
@@ -66,7 +66,7 @@ vcf =
     yarl
 bgen =
     rechunker
-    cbgen; platform_system != "Windows"
+    cbgen < 1.0.5; platform_system != "Windows"
     cbgen == 1.0.1; platform_system == "Windows"
 
 [coverage:report]

--- a/sgkit/stats/genedrop_numba_fns.py
+++ b/sgkit/stats/genedrop_numba_fns.py
@@ -167,7 +167,7 @@ def genedrop_Hamilton_Kerr(
     n_sample, _ = parent.shape
     _, max_ploidy = genotypes.shape
     order = topological_argsort(parent)
-    marked = np.zeros(max_ploidy, dtype=np.bool8)
+    marked = np.zeros(max_ploidy, dtype=np.bool_)
     np.random.seed(seed)
     for idx in range(n_sample):
         i = order[idx]

--- a/sgkit/stats/pedigree.py
+++ b/sgkit/stats/pedigree.py
@@ -412,16 +412,22 @@ def _hamilton_kerr_inbreeding_non_founder(
     """
     pat = (
         tau_p
-        * (tau_p - 1)
-        * (lambda_p + (1 - lambda_p) * ((ploidy_p * kinship_pp - 1) / (ploidy_p - 1)))
+        * (tau_p - 1.0)
+        * (
+            lambda_p
+            + (1.0 - lambda_p) * ((ploidy_p * kinship_pp - 1.0) / (ploidy_p - 1.0))
+        )
     )
     mat = (
         tau_q
-        * (tau_q - 1)
-        * (lambda_q + (1 - lambda_q) * ((ploidy_q * kinship_qq - 1) / (ploidy_q - 1)))
+        * (tau_q - 1.0)
+        * (
+            lambda_q
+            + (1.0 - lambda_q) * ((ploidy_q * kinship_qq - 1.0) / (ploidy_q - 1.0))
+        )
     )
-    num = pat + mat + 2 * tau_p * tau_q * kinship_pq
-    denom = tau_p * (tau_p - 1) + tau_q * (tau_q - 1) + 2 * tau_p * tau_q
+    num = pat + mat + 2.0 * tau_p * tau_q * kinship_pq
+    denom = tau_p * (tau_p - 1.0) + tau_q * (tau_q - 1.0) + 2.0 * tau_p * tau_q
     return num / denom
 
 

